File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/component/CafeRegionLabel.kt
package com.hhp227.concafe.presentation.component

import androidx.compose.runtime.Composable
import concafe.composeapp.generated.resources.Res
import concafe.composeapp.generated.resources.region_busan
import concafe.composeapp.generated.resources.region_daegu
import concafe.composeapp.generated.resources.region_osaka
import concafe.composeapp.generated.resources.region_seoul
import concafe.composeapp.generated.resources.region_tokyo
import concafe.composeapp.generated.resources.region_yokohama
import org.jetbrains.compose.resources.stringResource

@Composable
fun localizedRegionCity(rawCity: String): String {
    val normalized = rawCity.trim()
    if (normalized.isEmpty()) {
        return normalized
    }
    return when (normalized.lowercase()) {
        "seoul" -> stringResource(Res.string.region_seoul)
        "busan" -> stringResource(Res.string.region_busan)
        "daegu" -> stringResource(Res.string.region_daegu)
        "tokyo" -> stringResource(Res.string.region_tokyo)
        "osaka" -> stringResource(Res.string.region_osaka)
        "yokohama" -> stringResource(Res.string.region_yokohama)
        else -> normalized
    }
}

File: iosApp/iosApp/Presentation/Component/CafeRegionLabel.swift
//
//  CafeRegionLabel.swift
//  ConCafe
//
//  Created by 홍희표 on 7/20/26.
//

import Foundation

func localizedRegionCity(_ rawCity: String) -> String {
    let normalized = rawCity.trimmingCharacters(in: .whitespacesAndNewlines)
    guard !normalized.isEmpty else {
        return normalized
    }
    switch normalized.lowercased() {
    case "seoul":
        return String(localized: String.LocalizationValue("region_seoul"), table: "Localizable")
    case "busan":
        return String(localized: String.LocalizationValue("region_busan"), table: "Localizable")
    case "daegu":
        return String(localized: String.LocalizationValue("region_daegu"), table: "Localizable")
    case "tokyo":
        return String(localized: String.LocalizationValue("region_tokyo"), table: "Localizable")
    case "osaka":
        return String(localized: String.LocalizationValue("region_osaka"), table: "Localizable")
    case "yokohama":
        return String(localized: String.LocalizationValue("region_yokohama"), table: "Localizable")
    default:
        return normalized
    }
}
